utils/src/main/java/org/owasp/dependencycheck/utils/Checksum.java

```diff
@@ -79,21 +81,25 @@ private Checksum() {
      * not exist
      */
     public static byte[] getChecksum(String algorithm, File file) throws NoSuchAlgorithmException, IOException {
-        final MessageDigest md = MessageDigest.getInstance(algorithm);
-        try (FileInputStream fis = new FileInputStream(file);
-                FileChannel ch = fis.getChannel()) {
-            final ByteBuffer buf = ByteBuffer.allocateDirect(8192);
-            int b = ch.read(buf);
-            while (b != -1 && b != 0) {
-                buf.flip();
-                final byte[] bytes = new byte[b];
-                buf.get(bytes);
-                md.update(bytes, 0, b);
-                buf.clear();
-                b = ch.read(buf);
-            }
-            return md.digest();
+        HashFunction hashFunction = null;
+
+        switch (algorithm) {
```

Comment:
To ensure we don't run into casing issues should this be:

```suggestion
        switch (algorithm.toUpperCase()) {
```